MaterialLib/MPL/Medium.cpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \date   07.09.2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "Medium.h"
+
+#include <boost/variant.hpp>
+#include <string>
+
+#include "BaseLib/Algorithm.h"
+#include "Properties/Properties.h"
+
+#include "Component.h"
+#include "Phase.h"
+
+namespace MaterialPropertyLib
+{
+Medium::Medium(BaseLib::ConfigTree const& config)
```

Comment:
By 
1) making createProperties, createPhases be local functions only in this file, 
2) letting createProperties return PropertyArray, and let createPhases return std::vector<std::unique_ptr<Phase>> 
 
 ` _phases` and  `_properties` can be initialized in the initialization list.

Reply:
Followed your suggestion, though only `_phases` are initialized, since `_properties` are set in the ctor body.